File: backend/app/models/weather.py
import uuid
from sqlalchemy import Column, Integer, String, Float, DateTime, Boolean, UUID, Index
from sqlalchemy.sql import func
from app.core.database import Base


class Weather(Base):
    __tablename__ = "weather"

    __table_args__ = (
        Index('idx_weather_city', 'city'),
        Index('idx_weather_updated', 'updated_at'),
        Index('idx_weather_city_updated', 'city', 'updated_at'),
    )

    id = Column(UUID(as_uuid=True), primary_key=True, index=True, default=uuid.uuid4)
    city = Column(String(100), nullable=False, index=True)
    province = Column(String(100), nullable=True)
    country = Column(String(100), nullable=True)
    weather = Column(String(50), nullable=False)
    weather_img = Column(String(200), nullable=True)
    temperature = Column(String(20), nullable=False)
    temp_min = Column(String(20), nullable=True)
    temp_max = Column(String(20), nullable=True)
    wind_direction = Column(String(50), nullable=True)
    wind_speed = Column(String(20), nullable=True)
    wind_meter = Column(String(20), nullable=True)
    humidity = Column(String(20), nullable=True)
    visibility = Column(String(20), nullable=True)
    pressure = Column(String(20), nullable=True)
    air_quality = Column(String(50), nullable=True)
    is_daytime = Column(Boolean, default=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), index=True)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), index=True)

File: frontend/src/mock/weather.ts
import type { BackendWeatherData } from '@/services/backendWeatherService';

export const MOCK_WEATHER: BackendWeatherData = {
  city: '杭州',
  province: '浙江',
  country: '',
  weather: '多云',
  temperature: '24',
  tempMin: '18',
  tempMax: '28',
  windDirection: '东南风',
  windSpeed: '3级',
  humidity: '65%',
  visibility: '10km',
  pressure: '1012hPa',
  airQuality: '良',
  isDaytime: true,
  updateTime: new Date().toLocaleTimeString('zh-CN', { hour: '2-digit', minute: '2-digit' }),
};
